src/BenchmarksApps/TechEmpower/PlatformBenchmarks/BenchmarkApplication.Fortunes.cs

```diff
@@ -43,6 +45,37 @@ private void OutputFortunes(PipeWriter pipeWriter, List<Fortune> model)
             // Date header
             writer.Write(DateHeader.HeaderBytes);
 
+            var bodyStart = writer.Buffered;
+            // Body
+            writer.Write(_fortunesTableStart);
+            foreach (var item in model)
+            {
+                writer.Write(_fortunesRowStart);
+                writer.WriteNumeric((uint)item.Id);
+                writer.Write(_fortunesColumn);
+                HtmlEncoder.EncodeUtf8(item.Message.Span, writer.Span, out var bytesConsumed, out var bytesWritten, isFinalBlock: true);
+                Debug.Assert(bytesConsumed == item.Message.Length, "Not enough remaining space in the buffer");
+                writer.Advance(bytesWritten);
+                writer.Write(_fortunesRowEnd);
+            }
+            writer.Write(_fortunesTableEnd);
+            lengthWriter.WriteNumeric((uint)(writer.Buffered - bodyStart));
+
+            writer.Commit();
+        }
+
+        private void OutputFortunes(PipeWriter pipeWriter, List<FortuneDapper> model)
```

Comment:
Is it not possible to apply the same optimization using Dapper and/or EF Core?

Reply:
I tried this with Dapper, and unfortunately it doesn't support reading a string as a byte[]; it has its own type mapping system/constraints... It's something we can look into but it doesn't seem super critical either.

Similar with EF: the PG provider doesn't currently allow mapping a byte[] property to database `text`. Doing this seems maybe a bit extreme in a high-level ORM like EF, but we could...